Telegram/SourceFiles/mediaview.cpp

```diff
@@ -1204,8 +1204,8 @@ void MediaView::displayPhoto(PhotoData *photo, HistoryItem *item) {
 		if (auto photoMsg = dynamic_cast<HistoryPhoto*>(itemMsg->getMedia())) {
 			auto asBot = (item->author()->isUser()
 				&& item->author()->asUser()->botInfo);
-			auto skipw = qMax(_dateNav.left() + _dateNav.width(), _headerNav.left() + _headerNav.width());
-			auto maxw = qMin(qMax(width() - 2 * skipw - st::mediaviewCaptionPadding.left() - st::mediaviewCaptionPadding.right() - 2 * st::mediaviewCaptionMargin.width(), int(st::msgMinWidth)), _caption.maxWidth());
+			auto skipw = std::max(_dateNav.left() + _dateNav.width(), _headerNav.left() + _headerNav.width());
+			auto maxw = std::min(std::max(width() - 2 * skipw - st::mediaviewCaptionPadding.left() - st::mediaviewCaptionPadding.right() - 2 * st::mediaviewCaptionMargin.width(), int(st::msgMinWidth)), _caption.maxWidth());
```

Comment:
clamp

Reply:
Clamp cant be used here. There may be situations it will lead to undefined behavior, this constuction doesnt guarantee that second argument of std:max greater 0 and first argument of std::min greater 0, or one of these greater than other

From specs of std::clamp
"The behavior is undefined if the value of lo is greater than hi"